ubuntu_image/builder.py

```diff
@@ -412,51 +412,48 @@ def populate_filesystems(self):
 
     def _make_one_disk(self, imgfile, name, volume):
         part_id = 1
-        # XXX: This ought to be a single constructor that figures out the
-        # class for us when we pass in the schema.
-        if volume.schema is VolumeSchema.mbr:
-            image = MBRImage(imgfile, volume.image_size)
-        else:
-            image = Image(imgfile, volume.image_size)
+        # Create the image object for the selected volume schema
+        image = Image(imgfile, volume.image_size, volume.schema)
         offset_writes = []
         part_offsets = {}
+        # We first create all the partitions.
         for i, part in enumerate(volume.structures):
```

Comment:
It's a little difficult to tell from the diff, but do we need the `enumerate()` in this pass?  I know we need it below, but if we don't need it here, then we can just iterate over `volume.structures` and get rid of the `i`.